### PR TITLE
Require minimum of Kokkos version 3.4

### PR DESCRIPTION
**Pull request by dalg24**
Per our discussion we will commit to only support the last two versions of Kokkos.

- [ ] Update wiki

### CODE REVIEW

File: docker/README.md [hunk 1]
```diff
@@ -12,7 +12,7 @@ services:
   arborx_dev:
     build:
       args:
-        - BASE=nvidia/cuda:10.1-devel
-        - KOKKOS_VERSION=2.8.00
-        - KOKKOS_OPTIONS=--cxxstandard=c++14 --with-serial --with-openmp --with-options=disable_deprecated_code --with-cuda --with-cuda-options=enable_lambda --arch=SNB,Volta70
+        - BASE=nvidia/cuda:11.4.2-devel-ubuntu20.04
+        - KOKKOS_VERSION=3.5.00
+        - KOKKOS_OPTIONS=-DCMAKE_CXX_STANDARD=17 -DKokkos_ENABLE_OPENMP=ON -DKokkos_ENABLE_CUDA=ON -DKokkos_ENABLE_CUDA_LAMBDA=ON -DKokkos_ARCH_SNB=ON -DKokkos_ARCH_VOLTA70=ON
```
Comment (dalg24) [inline docker/README.md:17]: I decided to drop the enable Serial backend in the example snippet.  This is not an oversight.

**Review by masterleinad — APPROVED** [review]

**Comment by aprokop** [comment]
Weird PGI failures:
```
[ 15%] Linking CXX executable ArborX_BoostAdapters.exe

cd /var/jenkins/workspace/ArborX_PR-578/build/test && /opt/cmake/bin/cmake -E cmake_link_script CMakeFiles/ArborX_BoostAdapters.exe.dir/link.txt --verbose=1

/opt/pgi/linux86-64-llvm/19.4/bin/pgc++  -Minform=inform -g -O0  -DKOKKOS_DEPENDENCE -mp -Wl,-rpath "-Wl,\$$ORIGIN/../lib:\$$ORIGIN/../../lib:\$$ORIGIN/../../../lib:\$$ORIGIN/../../../../lib" -Wl,-rpath -Wl,/opt/pgi/linux86-64-llvm/2019/mpi/openmpi-3.1.3/lib CMakeFiles/ArborX_BoostAdapters.exe.dir/tstBoostGeometryAdapters.cpp.o CMakeFiles/ArborX_BoostAdapters.exe.dir/tstBoostRangeAdapters.cpp.o CMakeFiles/ArborX_BoostAdapters.exe.dir/utf_main.cpp.o  -o ArborX_BoostAdapters.exe  -Wl,-rpath,/opt/boost/lib:/opt/pgi/linux86-64-llvm/2019/mpi/openmpi-3.1.3/lib /opt/boost/lib/libboost_unit_test_framework.so.1.71.0 /opt/kokkos/lib/libkokkoscontainers.a /opt/kokkos/lib/libkokkoscore.a /opt/pgi/linux86-64-llvm/2019/mpi/openmpi-3.1.3/lib/libmpi_cxx.so /opt/pgi/linux86-64-llvm/2019/mpi/openmpi-3.1.3/lib/libmpi.so 

CMakeFiles/ArborX_CompileOnly.exe.dir/tstCompileOnlyAccessTraits.cpp.o: In function `Kokkos::Tools::Experimental::Impl::ValueHierarchyNode<long, Kokkos::Tools::Experimental::Impl::ValueHierarchyNode<long, void> >::~ValueHierarchyNode()':

/opt/kokkos/include/impl/Kokkos_ViewMapping.hpp:3184: undefined reference to `std::vector<Kokkos::Tools::Experimental::Impl::ValueHierarchyNode<long, void>, std::allocator<Kokkos::Tools::Experimental::Impl::ValueHierarchyNode<long, void> > >::~vector()'

/opt/kokkos/include/impl/Kokkos_ViewMapping.hpp:3184: undefined reference to `std::vector<long, std::allocator<long> >::~vector()'

```

**Comment by aprokop** [comment]
At the meeting, we agreed to remove the PGI build, and introduce nvhpc build at a later day. Removal of the PGI build should probably be done as a separate PR.

**Review by aprokop — APPROVED** [review]

**Comment by aprokop** [comment]
PGI build was removed. Everything else is fine.

**Comment by aprokop** [comment]
So this PR is functionally fine. However, it seems that it also blew up the SYCL build time from about 30 min to 1hr+.

**Pull request merged**